docs/ARCHITECTURE.md

```diff
@@ -787,6 +787,166 @@ direction LR
+            +__construct(?Psr18ClientInterface $client, ?Psr17RequestFactoryInterface $requestFactory, ?Psr17StreamFactoryInterface $streamFactory)
+            +send(Request $request) Response
+            -convertToPsr7Request(Request $request) Psr7RequestInterface
+            -convertFromPsr7Response(Psr7ResponseInterface $response) Response
+        }
+        
+        class HttpTransporterFactory {
+            +createTransporter() HttpTransporterInterface$
+        }
+    }
+    
+    namespace PSR {
+        class Psr17RequestFactoryInterface {
+            <<interface>>
+            +createRequest(string $method, UriInterface|string $uri) RequestInterface
+        }
+        
+        class Psr17StreamFactoryInterface {
+            <<interface>>
+            +createStream(string $content) StreamInterface
+        }
+        
+        class Psr18ClientInterface {
+            <<interface>>
+            +sendRequest(RequestInterface $request) ResponseInterface
+        }
+        
+        class Psr7RequestInterface {
+            <<interface>>
+        }
+        
+        class Psr7ResponseInterface {
+            <<interface>>
+        }
+    }
+    
+    <<interface>> HttpTransporterInterface
+    
+    HttpTransporter ..|> HttpTransporterInterface
+    HttpTransporter --> Psr17RequestFactoryInterface : uses
+    HttpTransporter --> Psr17StreamFactoryInterface : uses
+    HttpTransporter --> Psr18ClientInterface : uses
+    HttpTransporterFactory ..> HttpTransporter : creates
+    HttpTransporter ..> Request : receives
+    HttpTransporter ..> Response : creates
+    HttpTransporter ..> Psr7RequestInterface : creates
+    HttpTransporter ..> Psr7ResponseInterface : receives
+```
+
+### Integration with Models
+
+Models that need HTTP communication will use the `HttpTransporterInterface`:
+
+```mermaid
+---
+config:
+  class:
+    hideEmptyMembersBox: true
+---
+classDiagram
+    namespace AiClientNamespace.Providers.Models.Contracts {
+        class WithHttpTransporterInterface {
+            +setHttpTransporter(HttpTransporterInterface $transporter) void
+            +getHttpTransporter() HttpTransporterInterface
+        }
+    }
+    
+    namespace AiClientNamespace.Providers.Models.TextGeneration {
+        class SomeProviderTextGenerationModel {
+            -transporter HttpTransporterInterface
+            +generateTextResult(Message[] $prompt) GenerativeAiResult
+            -createApiRequest(Message[] $prompt) Request
+            -parseApiResponse(Response $response) GenerativeAiResult
+        }
+    }
+    
+    <<interface>> WithHttpTransporterInterface
+    
+    SomeProviderTextGenerationModel ..|> TextGenerationModelInterface
+    SomeProviderTextGenerationModel ..|> WithHttpTransporterInterface
+    SomeProviderTextGenerationModel --> HttpTransporterInterface : uses
+    SomeProviderTextGenerationModel --> Request : creates
+    SomeProviderTextGenerationModel --> Response : receives
+```
```

Comment:
Per https://github.com/WordPress/php-ai-client/pull/47#discussion_r2278021057, I'd say let's remove this portion to avoid too much detail for us to maintain:
```suggestion
```